From 516cb17f518e2f7720c5a069a8d141ff8378f0f6 Mon Sep 17 00:00:00 2001
From: Sergei Golovan <sgolovan@gmail.com>
Date: Sun, 9 May 2010 11:09:26 +0100
Subject: [PATCH] Fix Savannah bug #29828, test suite deadlock on FreeBSD.

* find/testsuite/find.gnu/sv-bug-24169.exp: Read find output until
eof before calling wait, in order to avoid a deadlock on FreeBSD.

Signed-off-by: James Youngman <jay@gnu.org>
---
 ChangeLog                                |    6 ++++++
 NEWS                                     |    4 ++++
 find/testsuite/find.gnu/sv-bug-24169.exp |    4 +++-
 3 files changed, 13 insertions(+), 1 deletions(-)

diff --git a/find/testsuite/find.gnu/sv-bug-24169.exp b/find/testsuite/find.gnu/sv-bug-24169.exp
index 33f96f8..349e823 100644
--- a/find/testsuite/find.gnu/sv-bug-24169.exp
+++ b/find/testsuite/find.gnu/sv-bug-24169.exp
@@ -23,7 +23,9 @@ set expected 1
 
 send_log "$cmd\n"
 eval spawn -noecho $cmd
-set result [wait]
+expect {
+    eof { set result [wait] }
+}
 send_log "result is '$result'\n"
 if { [ lindex $result 2 ] == 0  } then {
     set status [ lindex $result 3]
-- 
1.7.2.5

